registry/job.go

```diff
@@ -101,9 +101,21 @@ func (r *EtcdRegistry) Units() ([]job.Unit, error) {
 		return nil, err
 	}
 
+	// Fetch all units by hash recursively to avoid sending N requests to Etcd.
+	hashToUnit := r.getAllUnitsHashMap()
+	fetchUnitClosure := func(hash unit.Hash) *unit.UnitFile {
```

Comment:
because the interface of `dirToUnit` and all calls below need to be able to resolve either one hash->Unit or multiple ones.